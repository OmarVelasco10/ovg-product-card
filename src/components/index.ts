import { ProductCard as ProductCardHOC } from "./ProductCard";

import { ProductButtons } from "./ProductButtons";
import { ProductImage } from "./ProductImgae";
import { ProductTitle } from "./ProductTitle";
import { ProductCardHOCProps } from "../interfaces/interfaces";

export { ProductButtons } from "./ProductButtons";
export { ProductImage } from "./ProductImgae";
export { ProductTitle } from "./ProductTitle";

export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductButtons
})

export default ProductCard;

