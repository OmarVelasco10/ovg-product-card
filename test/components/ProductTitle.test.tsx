import renderer from 'react-test-renderer';
import * as React from 'react';
// import * as ReactDOM from 'react-dom';
import { ProductTitle, ProductCard } from '../../src/components/';
import { product } from '../data/product';

describe('ProductTitle test', () => {
    it('should to show the component with the custom title', () => {
      const wrapper = renderer.create(
        <ProductTitle title="Custom Product" className='custom-class'/>
      );

      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('Should to show the component with the product name', () => {
        const wrapper = renderer.create(
            <ProductCard product={product}>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
          );
    
          expect(wrapper.toJSON()).toMatchSnapshot();
    })
  });
  