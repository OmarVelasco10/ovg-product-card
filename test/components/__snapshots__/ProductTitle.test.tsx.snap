// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle test Should to show the component with the product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Sin imagen
  </span>
</div>
`;

exports[`ProductTitle test should to show the component with the custom title 1`] = `
<span
  className="productDescription custom-class"
>
  Custom Product
</span>
`;
